Initial pass at providing python packaging.

castorf opened the pull request:
This allows one to build a python package 'debi' for pypi.

castorf committed:
Initial pass at providing python packaging.

This allows one to build a python package 'debi' for pypi.

castorf commented:
Note that this moves things around quite a bit.  Chris's sharpcap scripts are now in src/debi/capture/SharpCap.

whkloepping commented:
Castor - sorry about being a bit slow on the uptake but I’m not sure what to do with this - it’s not clear how I should download or install or use the software. I’m thinking a ‘how to do it’ file may be in order for those of us who don’t understand how GitHub works.

Later...whk

> On Jul 5, 2023, at 2:54 PM, castorf ***@***.***> wrote:
> 
> 
> Note that this moves things around quite a bit. Chris's sharpcap scripts are now in src/debi/capture/SharpCap.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/8#issuecomment-1622403431>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWV44KM6GPYDPYJWABG46ELXOXBATANCNFSM6AAAAAAZ7MZ55U>.
> You are receiving this because you are subscribed to this thread.
>

castorf commented:
I'm not happy with the way it's packaged now, so I'm not going to document
it just yet beyond what's in the readme:

If you are already familiar with using git through something like VS Code,
or Git Desktop,
you can get the code from [
https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi.git](here).
```
winget install --id Git.Git -e --source winget
git clone https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi.git
```
to fetch the scripts to your system.

Once you've fetched the scripts, you can update them with `git pull`


-castor

On Wed, Jul 5, 2023 at 2:36 PM whkloepping ***@***.***> wrote:

> Castor - sorry about being a bit slow on the uptake but I’m not sure what
> to do with this - it’s not clear how I should download or install or use
> the software. I’m thinking a ‘how to do it’ file may be in order for those
> of us who don’t understand how GitHub works.
>
> Later...whk
>
> > On Jul 5, 2023, at 2:54 PM, castorf ***@***.***> wrote:
> >
> >
> > Note that this moves things around quite a bit. Chris's sharpcap scripts
> are now in src/debi/capture/SharpCap.
> >
> > —
> > Reply to this email directly, view it on GitHub <
> https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/8#issuecomment-1622403431>,
> or unsubscribe <
> https://github.com/notifications/unsubscribe-auth/AWV44KM6GPYDPYJWABG46ELXOXBATANCNFSM6AAAAAAZ7MZ55U>.
>
> > You are receiving this because you are subscribed to this thread.
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/8#issuecomment-1622558833>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABPQH3YJTVLMY6GOEWXNX7LXOXM4XANCNFSM6AAAAAAZ7MZ55U>
> .
> You are receiving this because you were assigned.Message ID:
> ***@***.***>
>

castorf commented:
> Castor - sorry about being a bit slow on the uptake but I’m not sure what to do with this - it’s not clear how I should download or install or use the software. I’m thinking a ‘how to do it’ file may be in order for those of us who don’t understand how GitHub works. Later...whk
> […](#)
> On Jul 5, 2023, at 2:54 PM, castorf ***@***.***> wrote: Note that this moves things around quite a bit. Chris's sharpcap scripts are now in src/debi/capture/SharpCap. — Reply to this email directly, view it on GitHub <[#8 (comment)](https://github.com/Dynamic-Eclipse-Broadcast-Initiative/debi/pull/8#issuecomment-1622403431)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AWV44KM6GPYDPYJWABG46ELXOXBATANCNFSM6AAAAAAZ7MZ55U>. You are receiving this because you are subscribed to this thread.

Hi Bill, this is actually a "code review"  for an intermediary step. Things that might be worth looking at are the description of the "deb project".  I didn't realize that this was actually a reply regarding the code review initially.

castorf commented on README.md:
```diff
@@ -16,11 +16,12 @@ Once you've fetched the scripts, you can update them with `git pull`
-
-* capture -- for scripts to assist in capturing images (e.g. SharpCap)
-* analysis -- image processing scripts
+We've set this up to produce a python 'package' called 'debi' which should
```
The package name 'debi' is not available in pypi, so that should be changed to something we can control.

castorf reviewed (commented):
Please change the pypi package name so it is unique.  Maybe debi-astro ?

castorf committed:
Initial pass at providing python packaging.

This allows one to build a python package 'debi' for pypi which
includes planetary-system-stacker so one could have one cpython
environment.
